orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/AbstractServerGroupTask.groovy

```diff
@@ -108,6 +110,17 @@ abstract class AbstractServerGroupTask extends AbstractCloudProviderAwareTask im
     operation
   }
 
+  Moniker convertMoniker(Stage stage) {
+    if (TargetServerGroup.isDynamicallyBound(stage)) {
+      TargetServerGroup tsg = TargetServerGroupResolver.fromPreviousStage(stage);
+      return  tsg.getMoniker() == null ?  MonikerHelper.friggaToMoniker(tsg.getName()) : tsg.getMoniker();
+    } else {
```

Comment:
don't need the `else {`

and since this is groovy, we can do:

`return MonikerHelper.monikerFromStage(stage, serverGroupName ?: asgName)`